diff -ur papi-3.7.0.orig/src/Rules.perfctr papi-3.7.0.patches/src/Rules.perfctr
--- papi-3.7.0.orig/src/Rules.perfctr	2009-08-24 21:21:29.000000000 +0100
+++ papi-3.7.0.patches/src/Rules.perfctr	2009-10-09 10:11:53.137683917 +0100
@@ -69,7 +69,7 @@
 endif
 
 CC_R    	:= $(CC) -pthread
-CC_SHR		:= $(CC) -fPIC -DPIC -shared -Wl,-soname -Wl,libpapi.so -Xlinker "-rpath" -Xlinker "$(LIBDIR)"
+CC_SHR		:= $(CC) -fPIC -DPIC -shared -Wl,-soname -Wl,libpapi.so.3 -Xlinker "-rpath" -Xlinker "$(LIBDIR)"
 
 ifeq (pgf77,$(F77))
 	FFLAGS      += -Wall -Mextend $(BITFLAGS)
diff -ur papi-3.7.0.orig/src/Rules.perfctr-pfm papi-3.7.0.patches/src/Rules.perfctr-pfm
--- papi-3.7.0.orig/src/Rules.perfctr-pfm	2009-08-24 21:21:29.000000000 +0100
+++ papi-3.7.0.patches/src/Rules.perfctr-pfm	2009-10-09 10:12:13.249680559 +0100
@@ -78,7 +78,7 @@
 endif
 
 CC_R    	:= $(CC) -pthread
-CC_SHR		:= $(CC) -fPIC -DPIC -shared -Wl,-soname -Wl,libpapi.so -Xlinker "-rpath" -Xlinker "$(LIBDIR)"
+CC_SHR		:= $(CC) -fPIC -DPIC -shared -Wl,-soname -Wl,libpapi.so.3 -Xlinker "-rpath" -Xlinker "$(LIBDIR)"
 
 ifeq (pgf77,$(F77))
 	FFLAGS      += -Wall -Mextend $(BITFLAGS)
diff -ur papi-3.7.0.orig/src/Rules.pfm2 papi-3.7.0.patches/src/Rules.pfm2
--- papi-3.7.0.orig/src/Rules.pfm2	2009-08-24 21:21:29.000000000 +0100
+++ papi-3.7.0.patches/src/Rules.pfm2	2009-10-09 10:12:38.552430433 +0100
@@ -17,7 +17,7 @@
 endif
 
 CC_R    	:= $(CC) -pthread
-CC_SHR		:= $(CC) -fPIC -DPIC -shared -Wl,-soname -Wl,libpapi.so -Xlinker "-rpath" -Xlinker "$(LIBDIR)"
+CC_SHR		:= $(CC) -fPIC -DPIC -shared -Wl,-soname -Wl,libpapi.so.3 -Xlinker "-rpath" -Xlinker "$(LIBDIR)"
 
 ifeq (pgf77,$(F77))
 	FFLAGS      += -Mextend
diff -ur papi-3.7.0.orig/src/Rules.pfm_pcl papi-3.7.0.patches/src/Rules.pfm_pcl
--- papi-3.7.0.orig/src/Rules.pfm_pcl	2009-10-09 10:06:54.326680322 +0100
+++ papi-3.7.0.patches/src/Rules.pfm_pcl	2009-10-09 10:12:45.255680805 +0100
@@ -16,7 +16,7 @@
 endif
 
 CC_R    	:= $(CC) -pthread
-CC_SHR		:= $(CC) -fPIC -DPIC -shared -Wl,-soname -Wl,libpapi.so -Xlinker "-rpath" -Xlinker "$(LIBDIR)"
+CC_SHR		:= $(CC) -fPIC -DPIC -shared -Wl,-soname -Wl,libpapi.so.3 -Xlinker "-rpath" -Xlinker "$(LIBDIR)"
 
 ifeq (pgf77,$(F77))
 	FFLAGS      += -Mextend
diff -ur papi-3.7.0.orig/src/Rules.pfm_pcl.x86 papi-3.7.0.patches/src/Rules.pfm_pcl.x86
--- papi-3.7.0.orig/src/Rules.pfm_pcl.x86	2009-08-24 21:21:29.000000000 +0100
+++ papi-3.7.0.patches/src/Rules.pfm_pcl.x86	2009-10-09 10:12:52.808555272 +0100
@@ -16,7 +16,7 @@
 endif
 
 CC_R    	:= $(CC) -pthread
-CC_SHR		:= $(CC) -fPIC -DPIC -shared -Wl,-soname -Wl,libpapi.so -Xlinker "-rpath" -Xlinker "$(LIBDIR)"
+CC_SHR		:= $(CC) -fPIC -DPIC -shared -Wl,-soname -Wl,libpapi.so.3 -Xlinker "-rpath" -Xlinker "$(LIBDIR)"
 
 ifeq (pgf77,$(F77))
 	FFLAGS      += -Mextend
